lib/hm/dig.rb

```diff
@@ -0,0 +1,21 @@
+class Hm
+  module Dig
+    DIGGABLE_CLASSES = [Hash, Array].freeze
+
+    def self.dig(what, *keys)
+      return what.dig(*keys) if what.respond_to?(:dig)
+
+      if diggable?(what)
```

Comment:
Let's do `diggable?(...) or fail TypeError, ...` here, then there would be less indentation, and less explicit `return`s for the rest of the method :)